src/AbstractMCMC.jl

```diff
@@ -399,4 +399,75 @@ function steps!(
     return Stepper(rng, model, s, kwargs)
 end
 
+##################################
+# Sample-until-convergence tools #
+##################################
+
+struct StopException <: Exception end
+
+"""
+    sample([rng::AbstractRNG, ]model::AbstractModel, s::AbstractSampler, is_done::Function; kwargs...)
+
+`sample` will continuously draw samples without defining a maximum number of samples until
+a convergence criteria defined by a user-defined function `is_done` returns `true`.
+
+`is_done` is a function `f` that returns a `Bool`, with the signature
+
+```julia
+f(rng::AbstractRNG, model::AbstractModel, s::AbstractSampler, transitions::Vector, iteration::Int; kwargs...)
+```
+
+`is_done` should return `true` when sampling should end, and `false` otherwise.
+"""
+function StatsBase.sample(
+    model::AbstractModel,
+    s::AbstractSampler;
+    kwargs...
+)
+    return sample(GLOBAL_RNG, model, s, kwargs...)
+end
+
+function StatsBase.sample(
+    rng::AbstractRNG,
+    model::AbstractModel,
+    sampler::AbstractSampler,
+    is_done::Function;
```

Comment:
```suggestion
    is_done;
```
to allow callable structs that are not subtypes of `Function`?